config/routes.rb

```diff
@@ -3,6 +3,9 @@
 Rails.application.routes.draw do
   devise_for :users
 
+  post 'authenticate', to: 'authentication#authenticate'
+  post '/auth/signup', to:'authentication#signup'
+
   resources :users, only: [:index, :show] do 
     resources :posts, only: [:index, :show, :new, :create] do
       resources :comments, only: [:create]
```

Comment:
It's often a common practice to add `/api` to URL or `api/v1` with version number in order to identify as an API based URL see [namespace](https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing) for reference